examples/karpenter/main.tf

```diff
@@ -160,6 +165,9 @@ module "karpenter" {
   cluster_name           = module.eks.cluster_name
   irsa_oidc_provider_arn = module.eks.oidc_provider_arn
 
+  # add SSM access for debugging
+  iam_role_additional_policies = [ "arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore" ]
```

Comment:
This is already attached below, so why is this necessary here?

Reply:
yes this confused me too as I was reading the karpenter docs and wondered why I wasn't able to connect with the session manager.
`policies` in the karpenter module attaches, in the example, the SSM role to the service account, so, the karpenter deployment needs this to create nodes. I tried removing the policies setting and saw that karpenter was unable to provision new nodes until I added it back. In the [migration document](https://karpenter.sh/v0.27.3/getting-started/migrating-from-cas/), the helm template/install step annotates the serviceaccount via a helm value.  I don't know why the SSM role is attached in the example when, in the module, the irsa aws_iam_policy_document seems to be trying to mirror the KarpenterControllerRole seen in the migration document. 

For  the iam_role_additional_policies, I suppose it could be considered optional - it's not necessary for karpenter to run in my estimation, but it's the recommended way to [troubleshoot karpenter](https://karpenter.sh/v0.27.3/troubleshooting/#node-notready), with `aws ssm start-session`. In the same [migration document](https://karpenter.sh/v0.27.3/getting-started/migrating-from-cas/#create-iam-roles), the SSM policy is attached again to the NodeRole policy and the NodeRole policy is attached to a NodeInstanceProfile and the NodeInstanceProfile is passed in as another value to the karpenter helm chart.

I can make it optional/comment it out if you'd like, I found the example provided confusing when I wasn't able to connect with SSM while the policy was provided already in one place in the example.

but now I'm wondering why arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore isn't established in the module instead of being needed to the users of the module, if it is indeed necessary, and whether the module aws_iam_policy_document in the module has diverged from karpenter upstream.